### DISCUSSION TITLE
example in the docs with typical IOS-style regexp to select a subsection

### DISCUSSION THREAD

**Discussion by dmfigol** [scrapli/scrapli_cfg#1] [Ideas]
It would be nice to have an example in the docs with typical IOS-style regexp to select a subsection of config for selective replacement, e.g.
```
BGP_PATTERN = re.compile(r"^router bgp \d+$(?:\n^\s+.*$)*\n!\n", flags=re.I | re.M)
```

**Comment by carlmontanari** [comment]
Im honestly suuuuuper nervous to do this... I want to provide the option to do the substitution stuff, but I dont want to field questions about it or have things in the docs that are right for one version but wrong for another and then have people open issues about stuff like that. I def get the value it would add for folks and it could help grease the skids for adoption but I dont know if its worth it in the long run from a maintenance perspective. Thoughts?

**Comment by carlmontanari** [comment]
I think I am inclined to *not* have this in the "official" docs as I dont want to deal w/ maintaining it/having it not work on different versions/platforms/etc.. That said, I think discussions is a decent place to store things like this where I wouldn't feel compelled to keep maintaining it, so if you or any future folks read this and think that would be useful that seems like a good "show and tell" discussion topic!
